tempesta_fw/http.c

```diff
@@ -160,35 +123,57 @@ tfw_http_prep_hexstring(char *buf, u_char *value, size_t len)
 	}
 }
 
+/*
+ * HTTP 302 response.
+ * The response redirects the client to the same URI as the original request,
+ * but it includes 'Set-Cookie:' header field that sets Tempesta sticky cookie.
+ */
 TfwHttpMsg *
-tfw_http_prep_302(TfwHttpMsg *hm, TfwStr *cookie)
+tfw_http_prep_302(TfwHttpMsg *hmreq, TfwStr *cookie)
 {
+#define S_302_PART_01	S_302 S_CRLF S_F_DATE
+#define S_302_PART_02	S_CRLF S_F_CONTENT_LENGTH "0" S_CRLF S_F_LOCATION
+#define S_302_PART_03	S_CRLF S_F_SET_COOKIE
+#define S_302_FIXLEN	SLEN(S_302_PART_01 S_V_DATE S_302_PART_02 S_302_PART_03)
+#define S_302_KEEP	S_CRLF S_H_CONN_KA
+#define S_302_CLOSE	S_CRLF S_H_CONN_CLOSE
 	size_t data_len = S_302_FIXLEN;
+	int conn_flag = hmreq->flags & __TFW_HTTP_CONN_MASK;
+	TfwHttpReq *req = (TfwHttpReq *)hmreq;
 	TfwHttpMsg *resp;
-	TfwHttpReq *req = (TfwHttpReq *)hm;
 	TfwMsgIter it;
-	TfwStr host;
 	TfwStr rh = {
 		.ptr = (TfwStr []){
 			{ .ptr = S_302_PART_01, .len = SLEN(S_302_PART_01) },
 			{ .ptr = *this_cpu_ptr(&g_buf), .len = SLEN(S_V_DATE) },
 			{ .ptr = S_302_PART_02, .len = SLEN(S_302_PART_02) }
 		},
-		.len = SLEN(S_302_PART_01 S_302_PART_02 S_V_DATE),
+		.len = SLEN(S_302_PART_01 S_V_DATE S_302_PART_02),
 		.flags = 3 << TFW_STR_CN_SHIFT
 	};
+	TfwStr crlfcrlf = { .ptr = S_CRLFCRLF, .len = SLEN(S_CRLFCRLF) };
+	TfwStr crlf_keep = { .ptr = S_302_KEEP, .len = SLEN(S_302_KEEP) };
+	TfwStr crlf_close = { .ptr = S_302_CLOSE, .len = SLEN(S_302_CLOSE) };
+	TfwStr host, *crlf = &crlfcrlf;
```

Comment:
Well, the variables are constant, so it seems there is no sense to allocate them on the stack on each function call and we can make them static.